modules/sso.php

```diff
@@ -1087,13 +1087,21 @@ public function get_user_data( $user_id ) {
 	 *
 	 * @since 6.9.0
 	 *
-	 * @param array $envelopes Array of JITM messages received after API call.
+	 * @param array  $envelopes    Array of JITM messages received after API call.
+	 * @param string $message_path The message path to ask for.
 	 *
 	 * @return array $envelopes New array of JITM messages. May now contain only one message, about SSO.
 	 */
-	public function inject_sso_jitm( $envelopes ) {
-		// Bail early if that's not the first time the user uses SSO.
-		if ( true != Jetpack_Options::get_option( 'sso_first_login' ) ) {
+	public function inject_sso_jitm( $envelopes, $message_path ) {
+		/*
+		 * Bail early if:
+		 * - the request does not originate from wp-admin main dashboard.
+		 * - that's not the first time the user uses SSO.
+		 */
+		if (
+			'wp:dashboard:admin_notices' !== $message_path
+			|| true != Jetpack_Options::get_option( 'sso_first_login' ) // phpcs:ignore WordPress.PHP.StrictComparisons.LooseComparison
```

Comment:
Non-blocking note, this was in the code before but do we need type conversion here? What values might `Jetpack_Options::get_option( 'sso_first_login' )` be?

Reply:
You're right, this could be simpler.

3f85104028468a53529a98aedbd32f46e76d2c56